src/code_08026044.c

```diff
@@ -436,3 +436,583 @@ void sub_08026FAC(struct Cutscene *arg0) {
         }
     }
 }
+
+void sub_08029D00(struct Cutscene *arg0);
+void sub_08027108(struct Cutscene *arg0) {
+    struct Background *bg;
+    gDispCnt = DISPCNT_OBJ_1D_MAP | DISPCNT_MODE_0;
+    gDispCnt |= DISPCNT_OBJ_ON | DISPCNT_BG1_ON;
+    arg0->flags &= ~0x7000;
+    arg0->flags &= ~0x38000;
+    arg0->flags &= ~0x1000000;
+    arg0->flags |= 0x2000000;
+    arg0->unk350 = 4;
+    arg0->unk2D8 = 0;
+    arg0->unk2DC = 0;
+    arg0->unk2E0 = 0;
+    arg0->unk2E2 = -0xc00;
+    bg = &arg0->unk194;
+    gBgScrollRegs[1][0] = 0;
+    gBgScrollRegs[1][1] = 0;
+    gBgCntRegs[1] = BGCNT_PRIORITY(2) | BGCNT_CHARBASE(0) | BGCNT_SCREENBASE(31);
+    BgInit(bg, 0x6000000, 0, 0x600F800, 0, 0, 0x2EC, 0, 0, 0, 0, 0x20, 0x20, 0, 0, 0, 9, 0, 0, 0x7fff, 0x7fff);
+    LZ77UnCompVram(gUnk_082D7850[0x2EC]->tileset, (u16 *)bg->unk4);
+    sub_08153060(bg);
+    arg0->unk0 = sub_08029D00;
+}
+
+void sub_080272F8(struct Cutscene *arg0);
+void sub_08027208(struct Cutscene *arg0) {
+    struct Sprite *spr1 = &arg0->unkF4;
+    arg0->unk33C = 0x7800;
+    arg0->unk340 = 0xdc00;
+    arg0->unk34A = -0x300;
+    SpriteInitNoTilesVram(spr1, 0x300, 0x390, 0xf, 0, 0xff, 0x10, 5, arg0->unk33C >> 8, arg0->unk340 >> 8, 0x2000);
+    arg0->flags &= ~0x50000;
+    arg0->flags |= 0x28000;
+    SpriteInitNoTilesVram(&arg0->unk11C, 0x2c0, 0x391, 0xf, 0, 0xff, 0x10, 0xf, arg0->unkF4.x, arg0->unkF4.y, 0x2000);
+    arg0->flags &= ~0x100000;
+    arg0->flags |= 0x280000;
+    m4aSongNumStart(SE_08D5EDA0);
+    arg0->unk0 = sub_080272F8;
+}
+
+void sub_08029D18(struct Cutscene *arg0);
+void sub_080272F8(struct Cutscene *arg0) {
+    arg0->unk340 += arg0->unk34A;
+    arg0->unk34A += 8;
+    if (arg0->unk34A > 0) {
+        arg0->unk34A = 0;
+    }
+
+    arg0->unkF4.y = arg0->unk340 >> 8;
+    arg0->unk11C.y = arg0->unkF4.y;
+    if (arg0->unk34A == 0) {
+        arg0->unk0 = sub_08029D18;
+    }
+}
+
+void sub_08029D38(struct Cutscene *arg0);
+void sub_08027350(struct Cutscene *arg0) {
+    s32 a;
+    arg0->unk340 += arg0->unk34A;
+    a = arg0->unk34A -= 8;
+    //TODO: I think this is fakematching
+    if (a * 0x10000 < -0x4000000) {
+        arg0->unk34A = -0x400;
+    }
+    arg0->unkF4.y = arg0->unk340 >> 8;
+    arg0->unk11C.y = arg0->unkF4.y;
+    if (arg0->unkF4.y < -0x20) {
+        arg0->unk0 = sub_08029D38;
+    }
+}
```

Comment:
```suggestion
void sub_08027350(struct Cutscene *arg0) {
    arg0->unk340 += arg0->unk34A;
    arg0->unk34A -= 8;
    if (arg0->unk34A < -0x400) {
        arg0->unk34A = -0x400;
    }
    arg0->unkF4.y = arg0->unk340 >> 8;
    arg0->unk11C.y = arg0->unkF4.y;
    if (arg0->unkF4.y < -0x20) {
        arg0->unk0 = sub_08029D38;
    }
}
```
We are using a slightly different version of agbcc